### PR TITLE
Add dynamic loading of optimizer

### PR DESCRIPTION
**Pull request by aramoto99**
1. Changed the way optimizers are specified in the Config file.This change eliminates the need for reinstallation when the optimzier code is changed.
- `grid` -> `aiaccel.optimizer.GridSearchOptimizer`
- `random` -> `aiaccel.optimizer.RandomSearchOptimizer`
- `sobol` -> `aiaccel.optimizer.SobolSearchOptimizer`
- `nelder-mead` -> `aiaccel.optimizer.NelderMeadSearchOptimizer`
- `tpe` -> `aiaccel.optimizer.TpeSearchoptimizer`
(e.g.)
https://github.com/aistairc/aiaccel/blob/55fcab89415cc91125fb89f717bbd2716157085a/examples/benchmark/config.yaml#L16

<br>
2. Optimizers not included in the aiaccel package can also be specified.In that case, must be added path.

~~~ bash
export PYTHONPATH="/hoge:$PYTHONPATH"
~~~

### CODE REVIEW

**Comment by yoshipon** [comment]
Thank you!
How about specifying the class name in the configuration to remain the original class name.
For example, `aiaccel.optimizer.grid.GridOptimizer`

File: aiaccel/optimizer/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        return self.opt.Optimizer
+
+    @singledispatchmethod
```
Comment (yoshipon) [inline aiaccel/optimizer/optimizer.py:30]: Is it really necessary to construct a generic function?

**Review by yoshipon — CHANGES_REQUESTED** [review]

**Comment by yoshipon** [comment]
Thank you for updating the PR.
It would be better to consider a simpler way to specify the standard optimizer (e.g., NelderMeadOptimizer).

**Comment by tkokada** [comment]
@aramoto99 What kinds of rules are defined for any user optimizers? I think you find them using following codes.

https://github.com/aistairc/aiaccel/blob/6b3963ddd003c0a4e0f0a3fdc7217f64a92f5d89/aiaccel/start.py#L512-L519

**Review by yoshipon — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**